contributors/design-proposals/dynamic-admission-control-configuration.md

```diff
@@ -0,0 +1,292 @@
+
+The schema is copied from 
+[#132](https://github.com/kubernetes/community/pull/132) with a few
+modifications.
+
+```golang
+type AdmissionControlConfiguration struct {
+    metav1.TypeMeta 
+
+    // validation will only allow one name in 1.7, it can be relaxed later. 
+    v1.ObjectMeta
+
+    // ResourceInitializers is a list of resources and their default initializers
+    ResourceInitializers []ResourceDefaultInitializer
+
+    ExternalAdmissionHooks []ExternalAdmissionHook
+}
+
+// Because the order of initializers matters, and each resource might need
+// differnt order, the ResourceDefaultInitializers are indexed by Resource.
+type ResourceDefaultInitializer struct {
+    // APIGroup of the resource. Because we want to use APIGroup + Resource as a
+    // merge key, we don't use a Resource struct. See muti-fields merge key
+    // [proposal](https://github.com/kubernetes/community/blob/master/contributors/design-proposals/multi-fields-merge-key.md).
+    APIGroup string
+
+    // Resource identifies the type of resource to be initialized that should be
+    // initialized
+    Resource string
+
+    // Initializers are the default names that will be registered to this resource
+    Initializers []Initializer
+}
+
+type Initializer struct {
+    // Name is the string that will be registered to the resource that needs
+    // initialization.
+    Name string
+
+    // FailurePolicy defines what happens if there is no initializer controller
+    // takes action. Allowed values are Ignore, or Fail. If "Ignore" is set, 
+    // apiserver removes initilizer from the initializers list of the resource
+    // if the timeout (30s) is reached; If "Fail" is set, apiserver returns 
+    // timeout error if the timeout is reached.
+    FailurePolicy FailurePolicyType
+}
+
+type FailurePolicyType string
+
+const (
+    Ignore FailurePolicyType = "Ignore"
+    // **optional** For 1.7, only "Ignore" is allowed. We can add "Fail" when
+    // the feature is more mature.
+    Fail FailurePolicyType = "Fail"
+)
+
+type ExternalAdmissionHook struct {
+    // Name of the AdmissionHook. It must be unique. It is used as the merge key.
+    Name string
+
+    // ClientConfig defines how to talk to the hook.
+    ClientConfig AdmissionHookClientConfig
+
+    // Rules describes what operations on what resources/subresources the webhook cares about.
+    // The webhook cares about an operation if it matches any Rule.
+    // Each rule must has a unique APIGroup.
+    Rules []Rule
+
+    // FailurePolicy defines how unrecognized errors from the admission endpoint are handled -
+    // allowed values are Ignore, Fail. Default value is Fail
+    FailurePolicy FailurePolicyType
+}
+
+type Rule struct {
+    // Verbs is the list of verbs this hook will be invoked on - POST, PUT, or *
+    // for all operations. Defaults to '*'.
+    Verbs []OperationType
+
+    // APIGroup is the API group the resources belong to. '*' is all groups.
+    APIGroup string 
```

Comment:
The choice of schema in RBAC wasn't random.  It gives a selection of power for the common cases:

 1. I have a resource that has moved groups.  Let me have a single rule for my deployments and I'll list two groups
 2. I have a group that contains multiple resources.  Let me have a single rule for batch and I'll list jobs and cronjobs
 3. I have a group that contains multiple resources, some with subresources.  Let me have a single rule for extensions and I'll list replicasets, replicasets/scale, and daemonsets.

Given how things move and how admission plugins are commonly tied to particular types, using a `[]string` for APIGroups seems more appropriate.

As for whether the same selection criteria makes sense for admission and RBAC, I would claim that it does.  You commonly want admission plugins to span a resource which may move groups, span an entire API group, span particular resources inside of a given group, and span everything.  It is more rare to want to subdivide resource access on verbs: create on Pods and update on ReplicaSets isn't a thing I've seen people want to do.  Those are scenarios that the RBAC rule tuples work well covering.